app/uk/gov/hmrc/gform/gform/NewFormController.scala

```diff
@@ -212,22 +212,54 @@ class NewFormController(
       implicit request => implicit l => cache =>
         val queryParams: QueryParams = QueryParams.fromRequest(request)
 
-        val formIdData = FormIdData.Plain(UserId(cache.retrievals), formTemplateId)
-        handleForm(formIdData, cache.formTemplate)(newForm(formTemplateId, cache, queryParams)) { form =>
-          cache.formTemplate.draftRetrievalMethod match {
-            case NotPermitted =>
-              fastForwardService.deleteForm(formTemplateId, cache.toAuthCacheWithForm(form, noAccessCode), queryParams)
-            case OnePerUser(ContinueOrDeletePage.Skip) | FormAccessCodeForAgents(ContinueOrDeletePage.Skip) =>
-              auditService.sendFormResumeEvent(form, cache.retrievals)
-              redirectContinue[SectionSelectorType.Normal](cache, form, noAccessCode)
-            case _ =>
-              auditService.sendFormResumeEvent(form, cache.retrievals)
-              val continueFormPage = new ContinueFormPage(cache.formTemplate, choice)
-              Ok(continue_form_page(frontendAppConfig, continueFormPage)).pure[Future]
-          }
-        }
+        for {
+          formTemplate <- getLatestFormTemplate(formTemplateId, cache.retrievals)
+          formIdData   <- Future.successful(FormIdData.Plain(UserId(cache.retrievals), formTemplate._id))
+          res <-
+            handleForm(formIdData, formTemplate)(
+              newForm(formTemplate._id, cache.copy(formTemplate = formTemplate), queryParams)
+            ) { form =>
+              formTemplate.draftRetrievalMethod match {
+                case NotPermitted =>
+                  fastForwardService.deleteForm(
+                    formTemplate._id,
+                    cache.toAuthCacheWithForm(form, noAccessCode),
+                    queryParams
+                  )
+                case OnePerUser(ContinueOrDeletePage.Skip) | FormAccessCodeForAgents(ContinueOrDeletePage.Skip) =>
+                  auditService.sendFormResumeEvent(form, cache.retrievals)
+                  redirectContinue[SectionSelectorType.Normal](
+                    cache.copy(formTemplate = formTemplate),
+                    form,
+                    noAccessCode
+                  )
+                case _ =>
+                  auditService.sendFormResumeEvent(form, cache.retrievals)
+                  val continueFormPage = new ContinueFormPage(formTemplate, choice)
+                  Ok(continue_form_page(frontendAppConfig, continueFormPage)).pure[Future]
+              }
+            }
+        } yield res
     }
 
+  private def getLatestFormTemplate(
```

Comment:
This method doesn't do what name suggest. I'm not sure the logic of this method is correct.
Why we simply don't get latest `FormTemplateId`  here? Why all this form existence checking?

Reply:
According to AC4 [GFORMS-1669](https://jira.tools.tax.service.gov.uk/browse/GFORMS-1669) if there is a saved form for any previous version, the form will be opened. I think we need to check whether a form exists or not for all previous versions.

Reply:
I agree with AC4. But for simplicity sake we should get latest formTemplate and rely on `GformConnector.maybeForm` to find any potential old form.

In other words we should go from latest (youngest) to the oldest formTemplate, rather than from oldest to youngest, as this solution do.